src/query/service/src/interpreters/interpreter_privilege_grant.rs

```diff
@@ -136,7 +137,14 @@ impl GrantPrivilegeInterpreter {
         // if the object's owner is None, it's considered as PUBLIC, everyone could access it
         let owner = user_mgr.get_ownership(tenant, owner_object).await?;
         if let Some(owner) = owner {
-            let can_grant_ownership = available_roles.iter().any(|r| r.name == owner.role);
+            // if object has ownership, but the owner role is not exists, set owner role to ACCOUNT_ADMIN,
+            // only account_admin can grant this object.
+            let role = if !user_mgr.exists_role(tenant, owner.role.clone()).await? {
+                BUILTIN_ROLE_ACCOUNT_ADMIN.to_string()
```

Comment:
how about moving the default `BUILTIN_ROLE_ACCOUNT_ADMIN` inside the `user_mgr.get_ownership` function in the L138?

the other places which calls `user_mgr.get_ownership` may also require this logic

Reply:
i'd prefer to keep the logic consistent across different places, the special case on logic is easier to be forgotten in the future, and might hide corner cases we did not recognize.

Reply:
it's nice to have this semantic described in a doc.